scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py

```diff
@@ -839,18 +839,35 @@ def _create_states(self, state_model, table_model, row_index=None):
         for row in rows:
             self.sans_logger.information("Generating state for row {}".format(row))
             if not self.is_empty_row(row):
-                try:
-                    state = gui_state_director.create_state(row)
-                    states.update({row: state})
-                except ValueError as e:
-                    self.sans_logger.error("There was a bad entry for row {}. Ensure that the path to your files has "
-                                           "been added to the Mantid search directories! See here for more "
-                                           "details: {}".format(row, str(e)))
-                    raise RuntimeError("There was a bad entry for row {}. Ensure that the path to your files has "
-                                       "been added to the Mantid search directories! See here for more "
-                                       "details: {}".format(row, str(e)))
+                row_user_file = table_model.get_row_user_file(row)
+                if row_user_file:
+                    user_file_path = FileFinder.getFullPath(row_user_file)
+                    if not os.path.exists(user_file_path):
+                        raise RuntimeError("The user path {} does not exist. Make sure a valid user file path"
+                                           " has been specified.".format(user_file_path))
+
+                    user_file_reader = UserFileReader(user_file_path)
+                    user_file_items = user_file_reader.read_user_file()
+
+                    row_state_model = StateGuiModel(user_file_items)
+                    row_gui_state_director = GuiStateDirector(table_model, row_state_model, self._facility)
+                    self._create_row_state(row_gui_state_director, states, row)
+                else:
+                    self._create_row_state(gui_state_director, states, row)
         return states
 
+    def _create_row_state(self, director, states, row):
+        try:
+            state = director.create_state(row)
+            states.update({row: state})
+        except ValueError as e:
+            self.sans_logger.error("There was a bad entry for row {}. Ensure that the path to your files has "
+                                   "been added to the Mantid search directories! See here for more "
+                                   "details: {}".format(row, str(e)))
+            raise RuntimeError("There was a bad entry for row {}. Ensure that the path to your files has "
+                               "been added to the Mantid search directories! See here for more "
+                               "details: {}".format(row, str(e)))
```

Comment:
Is there a reason this can't be put into a variable rather than copy and pasted?